libomnireader/include/xdrthing.h

```diff
@@ -7,25 +7,121 @@
 
 #include <cstring>
 #include <string>
+#include <cstdint>
 
 class XDRImpl;
 
 class XDRThing {
 
 public:
     XDRThing();
-
+    XDRThing(const XDRThing& me) { abort(); }
+    XDRThing& operator=(const XDRThing& me) { abort(); }
```

Comment:
Make `virtual` and then use `override` in subclass to make this a compiler error not a stop.

Reply:
Or you can just delete the methods if you want it to not be intitialisable as base class
```c++
    XDRThing(const XDRThing& me) { } = delete;
```